### ISSUE TITLE
Unexpected results returned from DAX Query using "Not In" operator (Query Builder)

### ISSUE THREAD

**Issue by ebeery11** [DaxStudio/DaxStudio#501]
**Describe the bug**
When generating DAX queries using the Query Builder, filters applied using the "Not In" operator appear to be overwritten by subsequent filters applied to a column from the same table.  My best guess is that this is due to some subtle DAX nuance specifically pertaining to the use of the EXCEPT, which unfortunately I'm not quite DAX-savvy enough to grasp at the moment (I'm sure it's covered _somewhere_ in Marco's Definitive Guide...).  

Even though I think this is just a nuance of the DAX language, since I would consider the behavior to be quite unexpected, and also not intuitively consistent (see Additional Context), I thought it was worth logging as a bug so that potentially a warning or some other prevention could be incorporated into the Query Builder user interface.

**To Reproduce**
1. Model with a single table (tbl_Sales), containing 3 columns (Customer, Sales, and Customer Type).
![image](https://user-images.githubusercontent.com/16659917/104106500-6181f680-5284-11eb-9d39-48e3ec592a11.png)
2. Create measure [TotalSales]: `TotalSales = SUM(tbl_Sales[Sales])`
3. Within DAX Studio Query builder, drag tblSales[Customer] and [TotalSales] into the Columns/Measures area.
![image](https://user-images.githubusercontent.com/16659917/104106771-a195a900-5285-11eb-8242-a55f5285f94b.png)

4. Add two filters: The first, tblSales[Customer] "Not In" {"A","B"}.  The second, tblSales[Customer Type] "Is Not" "Type 2"
![image](https://user-images.githubusercontent.com/16659917/104106776-a9554d80-5285-11eb-9ea1-1de0d2c16928.png)

5. Query generated is:
```
/* START QUERY BUILDER */
EVALUATE
SUMMARIZECOLUMNS(
    tbl_Sales[Customer],
    KEEPFILTERS( EXCEPT( ALL( tbl_Sales[Customer] ), TREATAS( {"A","B"}, tbl_Sales[Customer] ))),
    KEEPFILTERS( FILTER( ALL( tbl_Sales[Customer Type] ), tbl_Sales[Customer Type] <> "Type 2" )),
    "TotalSales", [TotalSales]
)
/* END QUERY BUILDER */
```
6. Run the query, results obtained are below:
![image](https://user-images.githubusercontent.com/16659917/104106788-cd189380-5285-11eb-9b0d-dd79ea5f13eb.png)

**Expected behavior**
The expected query result should be only a single row, showing Customer C, and Total Sales of 30, as Customer C is the only Customer fitting the criteria "Not In" {"A", "B"} AND Customer Type <> "Type 2".

**Screenshots**
![image](https://user-images.githubusercontent.com/16659917/104106817-1537b600-5286-11eb-847f-884cead4ede0.png)

**Additional context**
There were a couple additional scenarios I tested which caused me to be even more baffled by this DAX result.

- The same issue is not observed when using the "In" operator, which is what leads me to believe it's somehow related to the EXCEPT function.
![image](https://user-images.githubusercontent.com/16659917/104106848-6ba4f480-5286-11eb-8d63-5f312044a451.png)

- The issue disappears if tbl_Sales[Customer Type] is added to the output Column/Measures.
![image](https://user-images.githubusercontent.com/16659917/104106882-a3ac3780-5286-11eb-8098-617d0972e03c.png)
This one really stumped me - why would adding another field to the output of the SUMMARIZECOLUMNS function affect how the filters are applied?

**Comment by dgosbell** [comment]
Thanks for logging this. However this looks to me like it might be a bug in the Microsoft Tabular Engine. I've reached out to the product team to see if I can get confirmation of this. My understand is `EXCEPT( ALL( tbl_Sales[Customer] ), TREATAS( {"A","B"}, tbl_Sales[Customer] ))` should be logically equivalent to `FILTER( ALL( tbl_Sales[Customer] ), NOT( tbl_Sales[Customer] IN {"A","B"} ))`  and it seems that EXCEPT works as expected when used on it's own, but it stops working when used in conjunction with other filter conditions. Given the NOT IN format works correctly I've changed the query builder to generate that for the time being as this is similar to the syntax that Power BI generates when you try to build this sort of output. 

I also found that re-writing this query to use CALCULATETABLE and ADDCOLUMNS instead of SUMMARIZECOLUMNS also works correctly. So I think this might be some sort of bug in the combination of SUMMARIZECOLUMNS and EXCEPT.

```
EVALUATE
CALCULATETABLE (
    ADDCOLUMNS ( VALUES ( tbl_Sales[Customer] ), "TotalSales", [TotalSales] ),
    EXCEPT (
        ALL ( tbl_Sales[Customer] ),
        TREATAS ( { "A", "B" }, tbl_Sales[Customer] )
    ),
    FILTER (
        ALL ( tbl_Sales[Customer Type] ),
        tbl_Sales[Customer Type] <> "Type 2"
    )
)
```
